solr/solrj-zookeeper/src/test/org/apache/solr/client/solrj/impl/NodeValueFetcherTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.client.solrj.impl;
+
+import java.util.Set;
+import org.apache.solr.client.solrj.impl.SolrClientNodeStateProvider.RemoteCallCtx;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.response.CollectionAdminResponse;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.embedded.JettySolrRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class NodeValueFetcherTest extends SolrCloudTestCase {
+
+  private static final String COLLECTION =
+      NodeValueFetcherTest.class.getSimpleName() + "_collection";
+
+  private static int numShards;
+  private static int numReplicas;
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    // Metrics should be enabled to get them with snitch
+    System.setProperty("metricsEnabled", "true");
+
+    numShards = random().nextInt(3) + 3;
+    numReplicas = random().nextInt(2) + 2;
+    int numNodes = random().nextInt(2) + 2;
```

Comment:
It would be enough to vary between 2-3 nodes, instead of 2-4? (Goal: Faster tests)